src/lib/pubkey/ecdsa/ecdsa.cpp

```diff
@@ -158,7 +158,7 @@ AlgorithmIdentifier ECDSA_Signature_Operation::algorithm_identifier() const {
 secure_vector<uint8_t> ECDSA_Signature_Operation::raw_sign(const uint8_t msg[],
                                                            size_t msg_len,
                                                            RandomNumberGenerator& rng) {
-   BigInt m = BigInt::from_bytes_with_max_bits(msg, msg_len, m_group.get_order_bits());
+   BigInt m = m_group.mod_order(BigInt::from_bytes_with_max_bits(msg, msg_len, m_group.get_order_bits()));
```

Comment:
I looks like the `m_group.mod_order` below in line 179 can be removed with this change. 
```cpp
// Line 179
m = m_group.multiply_mod_order(m_b, m_group.mod_order(m));
```